Add binary scanner license to docs (#1595)
OpenLiberty/ci.maven: docs/dev.md

Start a Liberty instance in dev mode. This goal also invokes the `create`, `install-feature`, and `deploy` goals before starting the runtime. **Note:** This goal is designed to be executed directly from the Maven command line.

> N.B. starting in 3.6.1, dev mode invokes `generate-features` if the `generateFeatures` configuration parameter is set to `true` (default is `false`). **This goal will modify the source configuration directory of your application.** See [generate-features](generate-features.md) for details. With the auto-generation of features functionality turned on, dev mode has a runtime dependency on IBM WebSphere Application Server Migration Toolkit for Application Binaries, which is separately licensed under ILAN. Link to license: https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/license/wamt/en.html 

Reviewer: ```suggestion
 Starting in version 3.6.1, dev mode invokes then `generate-features` goal when the `generateFeatures` configuration parameter is set to `true`. **This goal modifies the source configuration directory of your application.**  See [generate-features](generate-features.md) for details. The default value for the `generateFeatures` parameter is `false`. When the auto-generation of features function is turned on, dev mode has a runtime dependency on IBM WebSphere Application Server Migration Toolkit for Application Binaries, which is separately licensed under ILAN. Link to license: https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/license/wamt/en.html 
```

Reviewer: What are the implications of modifying the source config dir? How is it modified? It seems like we are giving the users important information but it's not clear why they need to know it or what they should do about it

>  Starting in version 3.6.1, dev mode invokes the `generate-features` goal when the `generateFeatures` configuration parameter is set to `true`. **This goal modifies the source configuration directory of your application.** See [generate-features](generate-features.md) for details. The default value for the `generateFeatures` parameter is `false`. When auto-generation of features is turned on, dev mode has a runtime dependency on IBM WebSphere Application Server Migration Toolkit for Application Binaries, which is separately licensed under ILAN. Link to license: https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/license/wamt 

Reviewer 2: ```suggestion
 Starting in version 3.6.1, dev mode invokes the `generate-features` goal when the `generateFeatures` configuration parameter is set to `true`. **This goal modifies the source configuration directory of your application.** See [generate-features](generate-features.md) for details. The default value for the `generateFeatures` parameter is `false`. When auto-generation of features is turned on, dev mode has a runtime dependency on IBM WebSphere Application Server Migration Toolkit for Application Binaries, which is separately licensed under IBM License Agreement for Non-Warranted Programs. See license [here](https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/license/wamt).
